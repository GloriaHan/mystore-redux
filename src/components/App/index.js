import React, { useState } from "react";
import Routing from "../Routing";
export const CartContext = React.createContext();
export const InputContext = React.createContext();

export function App(props) {
  const [productsInCart, setProductsInCart] = useState([]);
  const [inputValue, setInputValue] = useState("");

  return ( 
    <InputContext.Provider value={{ inputValue, setInputValue }}>
      <CartContext.Provider value={{ productsInCart, setProductsInCart }}>
        <Routing />
      </CartContext.Provider>
      </InputContext.Provider>

  );
}

export default App;
