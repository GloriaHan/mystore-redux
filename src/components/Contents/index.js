import React from "react";
import { Outlet } from "react-router-dom";
import Sidebar from "../Sidebar";
import { Root } from "./Contents.style";

export default function Contents() {
  return (
    <Root>
      <Sidebar />
      <Outlet />
    </Root>
  );
}
