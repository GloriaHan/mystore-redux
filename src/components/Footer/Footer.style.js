import styled from "styled-components";
import Box from "@mui/material/Box";

export const Root = styled.div`
  width: 100%;
  background-color: #00acc1;
  color: white;
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const Img = styled.img`
  height: 40px;
  width: 40px;
  padding: 5px;
`;

export const Paymethods = styled.div`
  padding-right: 100px;
  display: flex;
  align-items: center;
`;

export const SocialMedia = styled.div`
  padding-right: 100px;
  display: flex;
  align-items: center;
`;

export const ContactUs = styled(Box)`
  padding-right: 80px;
  display: flex;
  align-items: center;
`;
