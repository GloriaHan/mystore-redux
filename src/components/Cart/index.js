import React, { useContext } from "react";
import { CartContext } from "../App/index";
import {
  Root,
  Th,
  Img,
  Title,
  Qty,
  Product,
  Price,
  Table,
  Tr,
  Td,
  Thead,
  Tbody,
  Header,
  Container,
  StyledButton,
} from "./Cart.style";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import Box from "@mui/joy/Box";
import Chip from "@mui/joy/Chip";
import ChipDelete from "@mui/joy/ChipDelete";

export default function Cart() {
  const { productsInCart, setProductsInCart } = useContext(CartContext);

  const productsQty =
    productsInCart.length > 0
      ? productsInCart.reduce((a, b) => a + b.qty, 0)
      : 0;

  const totalPrice =
    productsInCart.length > 0
      ? productsInCart.reduce((a, b) => a + b.qty * b.price, 0)
      : 0;

  const addToCart = (e, item) => {
    let result = productsInCart.find((itemObj) => itemObj.id === item.id);
    if (result) {
      result.qty = e.target.value;
      setProductsInCart([...productsInCart]);
    }
  };
  const deleteProduct = (e, item) => {
    const newProductsList = [...productsInCart].filter((itemObj) => {
      return itemObj.id !== item.id;
    });
    setProductsInCart(newProductsList);
  };
  return (
    <Root>
      <Header>Your shopping Cart : {productsQty} items </Header>
      <Container>
        <Table>
          <Thead>
            <Tr>
              <Th>Product</Th>
              <Th>Quantity</Th>
              <Th>Unit Price</Th>
              <Th>Sub-Total</Th>
              <Th></Th>
            </Tr>
          </Thead>
          <Tbody>
            {productsInCart.map((item) => (
              <Tr key={item.id}>
                <Td>
                  <Product>
                    <Img src={item.image} alt={item.title} />
                    <Title>{item.title}</Title>
                  </Product>
                </Td>
                <Td>
                  <Qty>
                    <FormControl sx={{ m: 1, minWidth: 120 }} size="small">
                      <InputLabel id="demo-select-small">Qty</InputLabel>
                      <Select
                        labelId="demo-select-small"
                        id="demo-select-small"
                        value={item.qty}
                        label="Qty"
                        onChange={(e) => addToCart(e, item)}
                      >
                        {new Array(20).fill(1).map((item, index) => (
                          <MenuItem key={index} value={index + 1}>
                            {index + 1}
                          </MenuItem>
                        ))}
                      </Select>
                    </FormControl>
                  </Qty>
                </Td>
                <Td>
                  <Price>${item.price}</Price>
                </Td>
                <Td>
                  <Price>
                    {`$${Number(item.qty * item.price).toFixed(2)}`}
                  </Price>
                </Td>
                <Td>
                  <Box sx={{ display: "flex", gap: 1, alignItems: "center" }}>
                    <Chip
                      onClick={(e) => deleteProduct(e, item)}
                      variant="outlined"
                      color="danger"
                      endDecorator={<ChipDelete />}
                    >
                      Delete
                    </Chip>
                  </Box>
                </Td>
              </Tr>
            ))}
          </Tbody>
        </Table>
      </Container>

      <Container className="checkout">
        <div>
          <StyledButton variant="contained">Checkout</StyledButton>
          <p>items: {productsQty}</p>
          <p>Post fee : Free</p>
        </div>
        <div>
          <p>Total: ${`$${Number(totalPrice).toFixed(2)}`}</p>
        </div>
      </Container>
    </Root>
  );
}
