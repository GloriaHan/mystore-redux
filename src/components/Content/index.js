import React, { useEffect, useState, useContext } from "react";
import { useParams, useSearchParams } from "react-router-dom";
import { Root, Img, Title, Product, Price,NoResult } from "./Content.style";
import { useNavigate } from "react-router-dom";
import Box from "@mui/material/Box";
import Paper from "@mui/material/Paper";
import Button from "@mui/material/Button";
import { CartContext, InputContext } from "../App/index";

export default function Content() {
  const [loading, setLoading] = useState(false);
  const [products, setProducts] = useState([]);
  const { category } = useParams();
  const navigate = useNavigate();
  const [searchParams] = useSearchParams();
  const search = searchParams.get("search");
  const { productsInCart, setProductsInCart } = useContext(CartContext);
  const { setInputValue } = useContext(InputContext);

  useEffect(() => {
    (async () => {
      const url = `${process.env.REACT_APP_BASE_URL}/products${
        category ? `/category/${category}` : ""
      }`;
      setLoading(true);
      const res = await fetch(url);
      const data = await res.json();
      setLoading(false);
      const filterData = search
        ? data.filter(
            (item) =>
              item.title.toUpperCase().includes(search.toUpperCase()) ||
              item.description.toUpperCase().includes(search.toUpperCase())
          )
        : data;
      setProducts(filterData);
    })();
  }, [category, search]);

  if (loading === true) return null;
  return (
    <Root>
      {products.length <= 0 ? (
        <NoResult>
          <h3>Sorry, nothing matched your search.</h3>
          <h3>Please search again.</h3>
        </NoResult>
      ) : (
        products &&
        products.map((item) => (
          <Box
            key={item.id}
            sx={{
              display: "flex",
              flexWrap: "wrap",
              "& > :not(style)": {
                m: 1,
                width: 195,
                height: 260,
              },
            }}
          >
            <Paper elevation={2}>
              <Product
                onClick={() => {
                  navigate(`/products/${item.id}`);
                  setInputValue("");
                }}
              >
                <Img src={item.image} alt={item.title} />
                <Title>{item.title}</Title>
                <Price>${item.price}</Price>
                <Button
                  variant="contained"
                  size="small"
                  onClick={(e) => {
                    e.stopPropagation();
                    navigate("/cart");
                    setInputValue("");
                    let result = productsInCart.find(
                      (product) => product.id === item.id
                    );
                    if (result) {
                      result.qty = result.qty + 1 * 1;
                      setProductsInCart([...productsInCart]);
                    } else {
                      setProductsInCart([
                        ...productsInCart,
                        { ...item, qty: 1 },
                      ]);
                    }
                  }}
                >
                  BUY NOW
                </Button>
              </Product>
            </Paper>
          </Box>
        ))
      )}
    </Root>
  );
}
