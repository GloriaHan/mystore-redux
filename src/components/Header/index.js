import React, { useContext } from "react";
import { useNavigate } from "react-router-dom";
import {
  Img,
  Root,
  SearchBar,
  StyledButton,
  ShoppingCart,
  CartButton,
} from "./Header.style";
import shoppinglogo from "../../assets/image/shoppinglogo.png";
import Paper from "@mui/material/Paper";
import InputBase from "@mui/material/InputBase";
import IconButton from "@mui/material/IconButton";
import SearchIcon from "@mui/icons-material/Search";
import Badge from "@mui/material/Badge";
import Box from "@mui/material/Box";
import { CartContext, InputContext } from "../App/index";

export default function Header() {
  const { productsInCart } = useContext(CartContext);
  const { inputValue, setInputValue } = useContext(InputContext);
  const navigate = useNavigate();
  const productsQty =
    productsInCart.length > 0
      ? productsInCart.reduce((a, b) => a + b.qty, 0)
      : 0;

  return (
    <Root>
      <div  onClick={() => {
              navigate("/");
              setInputValue("");
            }}>
        <Img src={shoppinglogo} alt="logo" />
      </div>
      <SearchBar>
        <Paper
          component="form"
          sx={{
            p: "2px 4px",
            display: "flex",
            alignItems: "center",
            width: 400,
          }}
        >
          <InputBase
            sx={{ ml: 1, flex: 1 }}
            placeholder="Buying makes us happy"
            inputProps={{ "aria-label": "Buying makes us happy" }}
            type="text"
            value={inputValue}
            onChange={(e) => setInputValue(e.target.value)}
          />
          <IconButton sx={{ p: "10px" }} aria-label="search">
            <SearchIcon />
          </IconButton>
        </Paper>

        <Box sx={{ "& button": { m: 1 } }}>
          <StyledButton
            onClick={() => navigate(`/mystore/products?search=${inputValue}`)}
            variant="contained"
            size="large"
          >
            search
          </StyledButton>
        </Box>
      </SearchBar>

      <CartButton  onClick={() => {
              navigate("/cart");
              setInputValue("");
            }}>
        <Badge badgeContent={productsQty} color="error">
          <ShoppingCart fontSize="large" />
        </Badge>
      </CartButton>
    </Root>
  );
}
