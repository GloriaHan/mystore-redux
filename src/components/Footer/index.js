import React from "react";
import { Root, Img, Paymethods, SocialMedia, ContactUs } from "./Footer.style";
import ThumbUpOffAltIcon from "@mui/icons-material/ThumbUpOffAlt";
import Button from "@mui/material/Button";
export default function Footer() {
  return (
    <Root>
      <Paymethods>
        <span>Payment methods: &nbsp;</span>
        <Img
          src="https://cdn-icons-png.flaticon.com/512/39/39196.png"
          alt="visa"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/512/349/349237.png"
          alt="JCB"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/512/349/349247.png"
          alt="paypal"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/128/493/493771.png"
          alt="AMEX"
        ></Img>
      </Paymethods>
      <SocialMedia>
        <span>Like us? &nbsp;</span>
        <ThumbUpOffAltIcon />
        &nbsp; &nbsp;
        <Img
          src="https://cdn-icons-png.flaticon.com/128/733/733547.png"
          alt="facebook"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/512/2111/2111463.png"
          alt="instagram"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/512/1409/1409937.png"
          alt="twitter"
        ></Img>
        <Img
          src="https://cdn-icons-png.flaticon.com/128/5968/5968812.png"
          alt="ticktok"
        ></Img>
      </SocialMedia>

      <ContactUs sx={{ "& button": { m: 1 } }}>
        <Button variant="contained" size="small">
          Contact Us
        </Button>
      </ContactUs>
    </Root>
  );
}
