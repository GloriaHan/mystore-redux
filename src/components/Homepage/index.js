import React from "react";
import { Outlet } from "react-router-dom";
import Header from "../Header";
import Footer from "../Footer";
import { Root, Content } from "./Homepage.style";

export default function Homepage() {
  return (
    <Root>
      <Header />
      <Content>
        <Outlet />
      </Content>
      <Footer />
    </Root>
  );
}
