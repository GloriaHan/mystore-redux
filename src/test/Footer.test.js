import React from "react";
import "jest-styled-components";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { InputContext, CartContext } from "../components/App";
import Footer from "../components/Header";

const productsInCart = [];

describe("Render Test", function () {
  it("case: expect rendering correct", () => {
    const { container } = render(
      <InputContext.Provider value={{}}>
        <CartContext.Provider value={{ productsInCart }}>
          <BrowserRouter>
            <Footer />
          </BrowserRouter>
        </CartContext.Provider>
      </InputContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
