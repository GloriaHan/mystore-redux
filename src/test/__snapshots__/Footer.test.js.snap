// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #0097a7;
}

.c1 {
  height: 50px;
  border-radius: 20%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  background-color: #3f51b5 !important;
}

.c3:disabled {
  color: white !important;
  background-color: grey !important;
}

.c4 {
  color: white;
}

<div>
  <div
    class="c0"
  >
    <div>
      <img
        alt="logo"
        class="c1"
        src="shoppinglogo.png"
      />
    </div>
    <div
      class="c2"
    >
      <form
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vm0r4s-MuiPaper-root"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
        >
          <input
            aria-label="Buying makes us happy"
            class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            placeholder="Buying makes us happy"
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="MuiBox-root css-69324s"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root c3 css-jh47zj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class=""
    >
      <span
        class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge c4 css-tzssek-MuiSvgIcon-root"
          data-testid="AddShoppingCartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError BaseBadge-badge BaseBadge-invisible css-zkqb44-MuiBadge-badge"
        />
      </span>
    </div>
  </div>
</div>
`;
